source/beamline/BioXAS/BioXASZebraPulseControl.cpp

```diff
@@ -181,38 +206,41 @@ void BioXASZebraPulseControl::setEdgeTriggerValue(int value)
 
 void BioXASZebraPulseControl::setDelayBeforeValue(double value)
 {
-	if (!delayBeforeControl_->withinTolerance(value)){
-
-		if (timeUnitsValueString() == "ms")
-			delayBeforeControl_->move(value*1000);
-
-		else if (timeUnitsValueString() == "s")
-			delayBeforeControl_->move(value);
-
-		else if (timeUnitsValueString() == "10s")
-			delayBeforeControl_->move(value/10);
-	}
+	if (!delayBeforeControl_->withinTolerance(value))
+		delayBeforeControl_->move(value);
 }
 
 void BioXASZebraPulseControl::setPulseWidthValue(double value)
 {
-	if (!pulseWidthControl_->withinTolerance(value)){
+	if (!pulseWidthControl_->withinTolerance(value))
+		pulseWidthControl_->move(value);
+}
+
+void BioXASZebraPulseControl::setTimeUnitsValue(int value)
+{
+	if (!timeUnitsControl_->withinTolerance(double(value)))
+		timeUnitsControl_->move(double(value));
+}
 
-		if (timeUnitsValueString() == "ms")
-			pulseWidthControl_->move(value*1000);
+void BioXASZebraPulseControl::setDelayTime(double newTime)
+{
 
-		else if (timeUnitsValueString() == "s")
-			pulseWidthControl_->move(value);
+	if (delayTime_ != newTime) {
+		delayTime_ = newTime;
+		updateDelayBeforeValue();
 
-		else if (timeUnitsValueString() == "10s")
-			pulseWidthControl_->move(value/10);
+		emit delayBeforeValueChanged(delayTime_);
 	}
 }
 
-void BioXASZebraPulseControl::setTimeUnitsValue(int value)
+void BioXASZebraPulseControl::setPulseTime(double newTime)
 {
-	if (!timeUnitsControl_->withinTolerance(double(value)))
-		timeUnitsControl_->move(double(value));
+	if (pulseTime_ != newTime) {
+		pulseTime_ = newTime;
+		updatePulseWidthValue();
+
+		emit pulseWidthValueChanged(pulseTime_);
```

Comment:
Good point!  Probably have the current signals for all the controls and separate ones where have our own abstraction.